docs/developer-docs/functionality/bitcoin/bitcoin.md

```diff
@@ -0,0 +1,219 @@
+
+Significant complexity of the implementation is in the area of securely resolving forks and protecting against various kinds of attacks. For example, in order to be able to resolve forks, the Bitcoin canister needs to maintain a certain number of Bitcoin blocks which are not yet absorbed into the maintained UTXO set, but UTXOs therein must be considered in addition for UTXO queries efficiently.
+
+**Submitting transactions**
+
+Canisters can submit Bitcoin transactions to the Bitcoin canister using the corresponding management canister API. Doing so queues the transactions for being submitted to the Bitcoin network. In every subnet round, Adapters obtain the pending transactions from the Bitcoin canister and queue them for being submitted asynchronously to the Bitcoin network. This leads to an efficient and quick distribution of transactions in the Bitcoin network as every replica of the subnet submits transactions via multiple connected nodes of the Bitcoin network.
+
+### Novel threshold ECDSA protocol
+
+Threshold ECDSA refers to the implementation of the ECDSA signature protocol using threshold cryptography. In a threshold ECDSA protocol, the private ECDSA key is secret shared between multiple parties and only an eligible quorum of the parties can generate a signature using their respective private key shares. The private key never exists in reconstructed form, but only in its secret-shared form. Key generation generates private key shares for the parties.
+
+The threshold ECDSA protocol implemented as part of the Internet Computer\'s chain key technology toolbox uses a single master private key from which keys for canisters can be derived using BIP-32-like key derivation. Each canister has one such root key derived using the canister id and an arbitrary number of additional canister keys can be derived using a backward-compatible extension of BIP-32.
+
+Threshold ECDSA on the Internet Computer will be rolled out to one signing subnet initially that will answer signing requests of canisters. The subnet enforces that only the canister that controls a key may request signatures with this key. All API calls go through Xnet traffic and thus incur some extra latency.
+
+Canisters can query their own or other canisters\' public keys, including further derived public keys of canisters. Canisters can request signatures with private keys they control, i.e., their root private key and derived private key. For requests of public keys or signatures with derived keys, a derivation path can be specified in the respective API.
+
+## Deployment Architecture
+
+The Bitcoin functionality will be activated on a single subnet of the IC and API calls from canisters to the Bitcoin API are routed via Xnet communication, thus extra latency is incurred. The feature may, if needed, in the future be additionally activated on (some) app subnets to avoid the additional Xnet latency and to scale out to be able to respond to more requests per time unit.
```

Comment:
```suggestion
The Bitcoin functionality will be activated on a single subnet of the IC and API calls from canisters to the Bitcoin API are routed via Xnet communication, thus extra latency is incurred. The feature may, if needed, in the future be additionally activated on (some) application subnets to avoid the additional Xnet latency and to be able to respond to more requests per time unit.
```